CRM/Contact/Import/Form/DataSource.php

```diff
@@ -54,9 +54,11 @@ public function preProcess() {
     //Test database user privilege to create table(Temporary) CRM-4725
     $errorScope = CRM_Core_TemporaryErrorScope::ignoreException();
     $daoTestPrivilege = new CRM_Core_DAO();
-    $daoTestPrivilege->query("CREATE TEMPORARY TABLE import_job_permission_one(test int) ENGINE=InnoDB");
-    $daoTestPrivilege->query("CREATE TEMPORARY TABLE import_job_permission_two(test int) ENGINE=InnoDB");
-    $daoTestPrivilege->query("DROP TEMPORARY TABLE IF EXISTS import_job_permission_one, import_job_permission_two");
+    $tempTable1 = CRM_Utils_SQL_TempTable::build()->getName();
```

Comment:
This feels a bit odd if we are actually creating tables (non temporary) why is the test here creating temporary tables

Reply:
Oh wow - that routine is kinda silly - that errorScope stuff should be a try-catch.

Anyway I tested this & it works

Reply:
Think about the importer pageflow:

1. You upload a large a CSV file
2. It shows a config screen -- where you can choose field-mappings
3. It shows progress as it imports each of the records

In step 1, you need to put the uploaded content somewhere that will endure long enough to run the other steps (maybe 5 min; maybe an hour; maybe more).  The importer uses `CREATE TABLE` to produce such a semi-durable storage place. `CREATE TEMPORARY TABLE` is not going to give you a place that lives long enough. (You could plausibly rewrite the importer to use other places for temporary storage -- like a cache record or civicrm.files -- but that's a hypothetical and broader proposition.)

This test is more informative if it's doing `CREATE TABLE` not `CREATE TEMPORARY TABLE` -- because those two may be stored in different places with different permissioning. For example: A MySQL user could have [permission](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#privileges-provided-summary) to `CREATE TEMPORARY TABLE` but *not* to`CREATE` new tables. (*It's very easy to imagine a sysadmin improvising those permissions because there are use-cases/apps where it's sensible to split those permissions.*) But in this case it would be a false-pass.